test/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -138,4 +138,40 @@ using ChainRules: level2partition, level3partition, chol_blocked_rev, chol_unblo
             @test chol_unblocked_rev(B̄, B, true) ≈ chol_blocked_rev(B̄, B, 10, true)
         end
     end
+    @testset "qr" begin
+        @testset "the thing" begin
+            for n in [4, 6, 10], m in [4, 6, 10]
+                X = randn(n, m)
+
+                if n < m
+                    @test_throws ArgumentError F, dX_pullback = rrule(qr, X)
+                    continue
+                end
+
+                F, dX_pullback = rrule(qr, X)
+                for p in [:Q, :R]
+                    Y, dF_pullback = rrule(getproperty, F, p)
+                    Ȳ = randn(size(Y)...)
+
+                    dself1, dF, dp = dF_pullback(Ȳ)
+                    @test dself1 === NO_FIELDS
+                    @test dp === DoesNotExist()
+
+                    dself2, dX = dX_pullback(dF)
+                    @test dself2 === NO_FIELDS
+                    X̄_ad = unthunk(dX)
+
+                    # Explicitely convert to Matrix since FiniteDifferences seem to
+                    # be broken for LinearAlgebra.QRCompactWYQ (infinite to_vec
+                    # recursion)
```

Comment:
Can an issue openned for this an linked here?

Reply:
Done here https://github.com/JuliaDiff/FiniteDifferences.jl/issues/114